pandas/io/json/_json.py

```diff
@@ -1063,7 +1066,9 @@ def __init__(
         self.dtype = dtype
 
         if date_unit is not None:
-            date_unit = date_unit.lower()
+            # avoid lowercasing "D" but ensure retrocompatibility for other units
+            if date_unit != "D":
```

Comment:
Why not just put a lowercase D in `_STAMP_UNITS`?

Reply:
I wanted to keep the exact same format passed to the underlying function